### PR TITLE
Add AddSite route to frontend

### PR DESCRIPTION
**Pull request by chromium-52**
## Why

[ClickUp Ticket](https://app.clickup.com/t/2mrderx)

Add an AddSite form to the frontend Admin page that allows admins and super-admins to add new sites individually.

## This PR

* Use two existing forms, `EditSiteForm` and `UpdateSiteForm`, to allow users to add a site and its site entry to the database.
* Have the `Form` component in `EditSiteForm` take up 100% of the available width.
* Dependent on https://github.com/Code-4-Community/speak-for-the-trees-backend-v2/pull/110

## Screenshots

Admin dashboard (desktop + Firefox): 
image

## Verification Steps

 
Visually verified that all components on the admin page and the `UpdateSiteForm` component on the site page were rendered as expected using Firefox and Chrome (on desktop and mobile). Attempted to add sites with invalid and missing field values to ensure that errors were thrown when invalid inputs are given and that the site is correctly created otherwise.

### CODE REVIEW

**Review by Tvcz — APPROVED** [review]
👍

File: src/components/forms/editSiteForm/index.tsx [hunk 1]
```diff
@@ -6,15 +6,20 @@ import { EditSiteRequest } from '../ducks/types';
...
 }
 
+const EditForm = styled(Form)`
+  width: 100%;
+`;
+
 const EditSiteForm: React.FC<EditSiteFormProps> = ({ formInstance }) => {
   return (
     <>
-      <Form name="basic" form={formInstance}>
```
Comment (SofieDunt) [inline src/components/forms/editSiteForm/index.tsx:17]: ❓ Does this still look okay in the other places (I think just site page) this form is used?

File: src/api/protectedApiClient.ts [hunk 2]
```diff
@@ -404,6 +407,12 @@ const getStewardshipReport = (): Promise<StewardshipReport> => {
   );
 };
 
+const addSite = (request: AddSiteRequest): Promise<void> => {
+  return AppAxiosInstance.post(ProtectedApiClientRoutes.ADD_SITE, request).then(
+    (res) => res.data,
+  );
+};
+
```
Comment (SofieDunt) [inline src/api/protectedApiClient.ts:415]: Can you add some tests for this?

File: src/containers/adminDashboard/index.tsx [hunk 3]
```diff
@@ -20,11 +28,50 @@ const EditUser = styled.div`
...
+      ...request,
+    };
+    ProtectedClient.addSite(addSiteRequest)
+      .then(() => {
+        editSiteForm.resetFields();
+        updateSiteForm.resetFields();
+        message.success('Site added!').then();
+      })
+      .catch((err) => message.error(err.response.data));
+  };
```
Comment (SofieDunt) [inline src/containers/adminDashboard/index.tsx:73]: 💭 I think it'd be good to pull out this function and write some tests for it

**Review by SofieDunt — COMMENTED** [review]
Looks great! Just one question and some room to add tests

File: src/api/protectedApiClient.ts (hunk 2 above)
Comment (rymaju) [reply src/api/protectedApiClient.ts:415]: ^

**Review by rymaju — COMMENTED** [review]
Sofie is right

**Review by rymaju — APPROVED** [review]
lgtm

File: src/api/test/protectedApiClient.test.ts [hunk 4]
```diff
@@ -1082,6 +1082,128 @@ describe('Protected API Client Tests', () => {
...
+          siteNotes: null,
+        };
+        const result = await ProtectedApiClient.addSite(
+          testBadAddSiteRequest,
+        ).catch((err) => err.response.data);
+
+        expect(result).toEqual(response);
+      });
+    });
+
```
Comment (SofieDunt) [inline src/api/test/protectedApiClient.test.ts:1206]: 💯

File: src/components/forms/ducks/types.ts [hunk 5]
```diff
@@ -92,3 +92,13 @@ export interface UpdateSiteRequest {
   readonly treeNotes: string | null;
   readonly siteNotes: string | null;
 }
+
+export interface AddSiteRequest extends UpdateSiteRequest {
```
Comment (SofieDunt) [inline src/components/forms/ducks/types.ts:96]: You can combine the `EditSiteRequest` and `UpdateSiteRequest` here instead

File: src/components/forms/ducks/types.ts [hunk 6]
```diff
@@ -92,3 +92,13 @@ export interface UpdateSiteRequest {
...
+  readonly blockId: number;
```
Comment (SofieDunt) [inline src/components/forms/ducks/types.ts:97]: `blockId` should be optional

**Review by SofieDunt — COMMENTED** [review]
Looks good! Just one ⛏️ with the types

**Review by SofieDunt — APPROVED** [review]
👍

**Pull request merged**